charts/substra-backend/templates/job-add-incoming-nodes.yaml

```diff
@@ -39,4 +39,24 @@ spec:
             value: {{ $.Release.Name }}-postgresql
           - name: PYTHONUNBUFFERED
             value: "1"
+      initContainers:
+      - name: wait-postgresql
+        image: jwilder/dockerize
+        command: ['dockerize', '-wait', 'tcp://{{ $.Release.Name }}-postgresql:5432']
+      - name: init-migrate
```

Comment:
Is it necessary to run the migrations in each job, considering they are already run by the deployment?

Reply:
yes, as they are triggered in parallel of the deployment

Reply:
What would you think of making the jobs (add-users, add-incoming, add-outgoing) initContainers of the deployment? 

It seems to me that it would be simpler (and less code!) for a tiny performance penalty (lack of parallelism). But I could be missing something :)

Reply:
Yes we can.
But I'm concerned. What is better in term of modularity?
Removing a job by removing its dedicated file or removing the part of this job from the deployment.
Furthermore, these jobs do not have to be run in a production setup, only the server deployment is needed. In a sense, we can have the server running without having these jobs finished before running it.
That's why I chose not to include them in the server deployment.

Reply:
Got it!

One more question: my understanding is that the migrations can now be triggered several times in parallel (i.e. a job triggers the migrations, and at the same time another job triggers the migrations).

Does the migration script have some kind of locking mechanism to prevent race conditions?

Reply:
You can launch multiple times the migrate command for applying migrations. It will always add the new ones.
Regarding race condition and locking mechanism, I don't know for sure. But I think there is one with pgsql by default with django